modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -172,10 +173,13 @@ public void processOpts() {
             supportingFiles.add(new SupportingFile("StringUtil.mustache", invokerFolder, "StringUtil.java"));
         }
 
-        supportingFiles.add(new SupportingFile("auth/HttpBasicAuth.mustache", authFolder, "HttpBasicAuth.java"));
-        supportingFiles.add(new SupportingFile("auth/ApiKeyAuth.mustache", authFolder, "ApiKeyAuth.java"));
-        supportingFiles.add(new SupportingFile("auth/OAuth.mustache", authFolder, "OAuth.java"));
-        supportingFiles.add(new SupportingFile("auth/OAuthFlow.mustache", authFolder, "OAuthFlow.java"));
+        // google-api-client doesn't use the Swagger auth, because it uses Google Credential directly (HttpRequestInitializer)
+        if (!"google-api-client".equals(getLibrary())) {
```

Comment:
I think this should go in the GoogleApiClientCodegen.java since it's only useful for you generator.

Reply:
Okay, I'll do that right now, sounds like a plan.

Reply:
FYI this is following the existing pattern. There actually aren't any subclasses of `JavaClientCodegen`; it uses if-statements on the `library` throughout to determine how the configuration of each library. I'd have to figure out how to support a new subclass, it looks like it uses reflection to instantiate the `JavaClientCodegen` with the classes listed in `/Users/charlescapps/Premise/projects/swagger-codegen-charlescapps/modules/swagger-codegen/src/main/resources/META-INF/services/io.swagger.codegen.CodegenConfig`